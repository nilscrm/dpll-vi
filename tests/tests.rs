use insta_cmd::assert_cmd_snapshot;
use rstest::rstest;
use std::{fs::File, path::{Path, PathBuf}, process::Command};

fn test_cnf_files(group_name: &str, path: &Path) {
  insta::with_settings!({
    snapshot_path => format!("snapshots/{}", group_name),
    filters => [(r"Performance\s+Time\s+.*\sms\s+Speed.*IPS", "[Performance...]")] },
  {
    let stdin = File::open(path).unwrap();
    assert_cmd_snapshot!(path.file_stem().unwrap().to_str().unwrap(), Command::new("vine").args(["run", "dpll/main.vi"]).arg("--breadth-first").stdin(stdin));
  });
}

#[rstest]
fn test_minimal(#[files("tests/minimal/*.cnf")] path: PathBuf) {
  test_cnf_files("minimal", &path);
}

#[rstest]
fn test_medium(#[files("tests/medium/*.cnf")] path: PathBuf) {
  test_cnf_files("medium", &path);
}

#[rstest]
fn test_satlib_sat(#[files("tests/uf5-218_first_10/*.cnf")] path: PathBuf) {
  test_cnf_files("uf5-218_first_10", &path);
}

#[rstest]
fn test_satlib_unsat(#[files("tests/uuf50-218_first_10/*.cnf")] path: PathBuf) {
  test_cnf_files("uuf50-218_first_10", &path);
}
