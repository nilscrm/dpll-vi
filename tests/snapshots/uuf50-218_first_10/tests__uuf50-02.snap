---
source: tests/tests.rs
info:
  program: vine
  args:
    - run
    - dpll/main.vi
    - "--breadth-first"
---
success: true
exit_code: 0
----- stdout -----
unsatisfiable

----- stderr -----

Interactions
  Total           71_445_590
    Depth         36_681_544
    Breadth                1
  Annihilate      40_344_984
  Commute            100_400
  Copy             6_817_200
  Erase            8_160_302
  Expand           5_683_427
  Call             6_542_870
  Branch           3_796_407

Memory
  Heap               670_416 B
  Allocated    1_583_551_968 B
  Freed        1_583_551_968 B

[Performance...]
